osu.Game/Screens/Play/HUD/GameplayLeaderboard.cs

```diff
@@ -46,42 +75,114 @@ protected override void LoadComplete()
         {
-            throw new NotSupportedException($"Use {nameof(AddPlayer)} instead.");
+            base.Update();
+
+            if (requiresScroll && trackedScore != null)
+            {
+                float scrollTarget = scroll.GetChildPosInContent(trackedScore) + trackedScore.DrawHeight / 2 - scroll.DrawHeight / 2;
+                scroll.ScrollTo(scrollTarget, false);
+            }
+
+            const float panel_height = GameplayLeaderboardScore.PANEL_HEIGHT;
+
+            float fadeBottom = scroll.Current + scroll.DrawHeight;
+            float fadeTop = scroll.Current + panel_height;
+
+            if (scroll.Current <= 0) fadeTop -= panel_height;
+            if (!scroll.IsScrolledToEnd()) fadeBottom -= panel_height;
+
+            // logic is mostly shared with Leaderboard, copied here for simplicity.
+            foreach (var c in Flow.Children)
+            {
+                float topY = c.ToSpaceOfOtherDrawable(Vector2.Zero, Flow).Y;
+                float bottomY = topY + panel_height;
+
+                bool requireTopFade = requiresScroll && topY <= fadeTop;
+                bool requireBottomFade = requiresScroll && bottomY >= fadeBottom;
+
+                if (!requireTopFade && !requireBottomFade)
+                    c.Colour = Color4.White;
+                else if (topY > fadeBottom + panel_height || bottomY < fadeTop - panel_height)
+                    c.Colour = Color4.Transparent;
+                else
+                {
+                    if (bottomY - fadeBottom > 0 && requiresScroll)
+                    {
+                        c.Colour = ColourInfo.GradientVertical(
+                            Color4.White.Opacity(Math.Min(1 - (topY - fadeBottom) / panel_height, 1)),
+                            Color4.White.Opacity(Math.Min(1 - (bottomY - fadeBottom) / panel_height, 1)));
+                    }
+                    else if (requiresScroll)
+                    {
+                        c.Colour = ColourInfo.GradientVertical(
+                            Color4.White.Opacity(Math.Min(1 - (fadeTop - topY) / panel_height, 1)),
+                            Color4.White.Opacity(Math.Min(1 - (fadeTop - bottomY) / panel_height, 1)));
+                    }
+                }
+            }
         }
 
         private void sort()
         {
             if (sorting.IsValid)
                 return;
 
-            var orderedByScore = this.OrderByDescending(i => i.TotalScore.Value).ToList();
+            var orderedByScore = Flow.OrderByDescending(i => i.TotalScore.Value).ToList();
 
-            for (int i = 0; i < Count; i++)
+            for (int i = 0; i < Flow.Count; i++)
             {
-                SetLayoutPosition(orderedByScore[i], i);
+                Flow.SetLayoutPosition(orderedByScore[i], i);
                 orderedByScore[i].ScorePosition = i + 1;
             }
 
             sorting.Validate();
         }
+
+        private class ManualScrollScrollContainer : OsuScrollContainer
```

Comment:
The double "scroll" in this name seems a little funny.

Reply:
Have renamed to hopefully something more appropriate and less funny.